### PR TITLE
Adding new data for gas metallicity plots

### PR DESCRIPTION
**Pull request by correac**
New datasets from Hayden-Pawson+ (2022), Nicholls+ (2017) and Berg+ (2020) to complement the new gas phase C/O-O/H and N/O-O/H relations.

### CODE REVIEW

**Comment by MatthieuSchaller** [comment]
Note that the checks are failing. See above.

**Comment by correac** [comment]
@MatthieuSchaller The error occurs in > 
`Script data/GalaxyH2MassFunction/conversion -> "python3 -W error -W ignore::UserWarning convertFletcher2021.py ../../../cosmology.py" failed.`
I'm not sure why that happens. I did not modify anything other than the data in folder GalaxyStellarMassGasMetallicity.

File: data/GalaxyStellarMassGasMetallicity/conversion/convertHayden2022.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    logOH_median, scatter=None, comoving=True, description="Gas-phase metallicity"
+)
+processed.associate_y(
+    logNO_median,
+    scatter=y_scatter,
+    comoving=True,
+    description="Gas-phase nitrogen over oxygen",
+)
+processed.associate_citation(citation, bibcode)
+processed.associate_name(name)
```
Comment (robjmcgibbon) [inline data/GalaxyStellarMassGasMetallicity/conversion/convertHayden2022.py:101]: You don't redefine name, it's still set as "Stellar mass - Gas phase N/O relation"

**Review by robjmcgibbon — CHANGES_REQUESTED** [review]
https://github.com/SWIFTSIM/velociraptor-comparison-data/pull/181/ fixes the issue that is currently causing the check to fail. Please rebase the master branch

**Comment by MatthieuSchaller** [comment]
Thanks Rob!

**Comment by MatthieuSchaller** [comment]
Is this still [WIP]?

File: data/GalaxyStellarMassGasMetallicity/conversion/convertHayden2022.py (hunk 1 above)
Comment (correac) [reply data/GalaxyStellarMassGasMetallicity/conversion/convertHayden2022.py:103]: Fixed!

**Comment by correac** [comment]
New datasets from Hayden-Pawson+ (2022), Nicholls+ (2017) and Berg+ (2020) to complement the gas phase C/O-O/H and N/O-O/H relations. The data is shown below. @MatthieuSchaller and @robjmcgibbon you can review the changes, for me this PR is ready to be reviewed and merged.

image
image

**Review by robjmcgibbon — APPROVED** [review]

**Pull request merged**